src/Components/Components/src/ChangeEventArgs.cs

```diff
@@ -10,9 +10,24 @@ namespace Microsoft.AspNetCore.Components
     /// </summary>
     public class ChangeEventArgs : EventArgs
     {
+        private string?[]? _multipleValues;
+
         /// <summary>
         /// Gets or sets the new value.
         /// </summary>
         public object? Value { get; set; }
+
+        /// <summary>
+        /// Gets or sets the new array of values.
+        /// </summary>
+        public string?[]? MultipleValues
+        {
+            get => _multipleValues;
+            set
+            {
+                _multipleValues = value;
+                Value = value;
+            }
+        }
```

Comment:
We populate `Value` here as well because value bindings are implemented via the `onchange` event and depend on `Value` to work properly.

We could alternatively look into a way to make `Value` of type `string[]` in the `onchange` event for multiple select.

Reply:
> We could alternatively look into a way to make Value of type string[] in the onchange event for multiple select.

If that's possible it would certainly be cleaner (as people might wonder what `MultipleValues` means in other cases). Do you have a sense of how difficult it would be?

Reply:
TBH it looks like you'd just change the line at https://github.com/dotnet/aspnetcore/pull/33950/files#diff-69a55d3b3e3176f789e85bd0ae14691fe3c51ad6eaaf3df7a7c76366bc273f88R256, but maybe I'm missing reasons why it's more complicated.